Temporal Cloud limits (#2277)
temporalio/documentation: docs-src/cloud/temporal-cloud-limits-sheet.md

---
slug: temporal-cloud-limits-sheet
title: What are the Temporal Cloud default limits?
sidebar_label: Temporal Cloud default limits
description: This section describes many of the default settings and limits that apply to Workflow Executions in Temporal Cloud.
tags:
  - error
  - warn
  - limits
---

This section describes many of the default settings and limits that apply to Workflow Executions and do not pertain to the entire Temporal Cloud account or individual Namespaces in Temporal Cloud, unless otherwise specified.

> Some of these default limits are configurable by sending a [support ticket](/cloud/support#support-ticket) to our support team.

Reviewer: ```suggestion
Some of these default limits are configurable by sending a [ticket](/cloud/support#support-ticket) to our support team.
```

repetition of "support" felt weird

### What is the maximum length for identifiers?

Identifiers such as Workflow Id, Workflow Type, and Task Queue names have a maximum length of 1000 characters in UTF-8 format.

> This default limit is configurable by creating a [support ticket](/cloud/support#support-ticket).

Reviewer: is this true? @tsurdilo

### What is the gRPC limit for each message received?

Each gRPC message received has a limit of 4 MB.

> ### What is the maximum limit for an Event batch size?
> 
> The `DefaultTransactionSizeLimit` is configured at 4 MB, representing the maximum allowable transaction size for persisting Event Histories.

Reviewer: ```suggestion
### What is the limit for History transaction size?

The `DefaultTransactionSizeLimit` is configured at 4 MB, representing the maximum allowable transaction size for persisting Event Histories.
```

### What are the default limits at the Namespace level?

At the Namespace level, Temporal Cloud sets the following default limits:

- Actions per second: 200 (with spikes to 400)
- Certificates: 32 KB payload or 16 certificates, whichever is smaller
- Concurrent Task pollers: 2,000 (configurable; maximum of 100,000)

> ### What is the Concurrent limit?
> 
> If the concurrent pending count of certain Commands exceeds 2,000, they will fail.
> These commands include:
> 
> - `ScheduleActivityTask`
> - `SignalExternalWorkflowExecution`
> - `RequestCancelExternalWorkflowExecution`
> - `StartChildWorkflowExecution`
> 
> This default limit is configurable by creating a [support ticket](/cloud/support#support-ticket).

Reviewer: ```suggestion
### What are the concurrecy limits?

If the concurrent pending count of the following Commands exceeds 2,000, additional Commands of that type will fail:

- `ScheduleActivityTask`
- `SignalExternalWorkflowExecution`
- `RequestCancelExternalWorkflowExecution`
- `StartChildWorkflowExecution`

These default limits are configurable by creating a [support ticket](/cloud/support#support-ticket).

[Batch jobs](/cli/batch) are limited to one job at a time.
```

removed "include" because it's a complete list. reworded because "they will fail" could be applied to the 2,000. also plural limits because each is a different setting and can be changed separately. added batch jobs because that's another thing we have a concurrency limit on. put it after support ticket because i don't know if it's configurable.

### What is the limit on the number of Signals received per Workflow Execution?

> The number of Signals received per Workflow Execution to 10,000.

Reviewer: ```suggestion
10,000
```

If there's a question, I'd rather just read the answer than repeating the text. (same for other questions)

If we keep the full-sentence answer, it should have "is limited to"

### What are the default limits for users and Namespaces at the account level?

At the account level, Temporal Cloud sets the following default limits:

- Users: 100
- Namespaces: 10

> ### What is the limit for Prometheus endpoint Retention Period?
> 
> The Prometheus endpoint Retention Period is 30 days.

Reviewer: is this the same retention period as workflow Retention Period? @tsurdilo do you know? and does it mean # of days of data i can see when hitting the endpoint? 

if changing the workflow Retention Period doesn't affect Prometheus, i'd change to:

```suggestion
### How much data does the Prometheus endpoint retain?

The Prometheus endpoint retains 30 days of data.
```

### What is the default Retention Period in your Temporal Cloud Account?

> Temporal Cloud sets the default Retention Period to 30 days and is configurable in the Temporal Web UI per Namespace.

Reviewer: ```suggestion
Temporal Cloud sets the default [Retention Period](/clusters#retention-period) to 30 days, and it is configurable in the Temporal Web UI per Namespace.
```

> You can set the Retention Period to a range of 1—90 days.

Reviewer: ```suggestion
You can set the Retention Period to between 1 and 90 days.
```

It's set to a discrete number, not a range

> ### What are the default limits for Custom Search Attributes?
> 
> The default limits for Custom Search Attributes in Temporal Cloud (maximum per type) are:
> 
> - `bool`: 20
> - `double`: 20
> - `datetime`: 20
> - `int`: 20
> - `keyword`: 20
> - `text`: 5

Reviewer: ```suggestion
### What are the default maximum numbers of Custom Search Attributes?

The default maximum number of [Custom Search Attributes](/visibility#custom-search-attributes) of each type in Temporal Cloud are:

- `bool`: 20
- `double`: 20
- `datetime`: 20
- `int`: 20
- `keyword`: 20
- `text`: 5
```

---
slug: temporal-cloud-limits-sheet
title: What are the Temporal Cloud default limits?
sidebar_label: Temporal Cloud default limits
description: This section describes many of the default settings and limits that apply to Workflow Executions in Temporal Cloud.
tags:
  - error
  - warn
  - limits
---

> This section describes many of the default settings and limits that apply to Workflow Executions and do not pertain to the entire Temporal Cloud account or individual Namespaces in Temporal Cloud, unless otherwise specified.

Reviewer 2: This keeps tripping me up - 

I think we should specify at each default limit whether it applies to a Workflow Execution, a Namespace, or an Account.

```suggestion
This section describes many of the default settings and limits that apply to Workflow Executions, Namespaces, and Accounts while using Temporal Cloud.
```

---
slug: temporal-cloud-limits-sheet
title: What are the Temporal Cloud default limits?
sidebar_label: Default limits
description: This section describes many of the default settings and limits that apply to Workflow Executions in Temporal Cloud.
tags:
  - error
  - warn
  - limits
---

> This section describes many of the default settings and limits that apply application operations in Temporal Cloud.

Reviewer 2: ```suggestion
This section describes many of the default settings and limits that apply to application operations in Temporal Cloud.
```

Some of these default limits are configurable by sending a [ticket](/cloud/support#support-ticket) to our support team.

### What is the maximum length for identifiers?

Identifiers such as Workflow Id, Workflow Type, and Task Queue names have a maximum length of 1,000 characters in UTF-8 format.

This default limit is configurable by creating a [support ticket](/cloud/support#support-ticket).

### What is the gRPC limit for each message received?

Each gRPC message received has a limit of 4 MB.

> This limit applies to the entire Temporal platform at every endpoint.

Reviewer 2: ```suggestion
This limit applies to all gRPC endpoints across the Temporal Platform.
```
